File: R/read.R
#' @title read_csv_file
#'
#' @description Read a csv datafile
#'
#' @param fpath path of the csv file
#' @export
read_csv_file <- function (fpath) {
    return(read.csv(
        fpath,
        sep=',',
        quote='"',
        encoding='utf-8'
    ))
}

#' @title read_yaml_file
#'
#' @description read YAML config file
#'
#' @param fpath path of the YAML file
#' @export
read_yml_file <- function (fpath) {
    fhandle <- file(fpath, "r", encoding="utf-8")
    ret <- yaml::read_yaml(file=fhandle, fileEncoding="utf-8")
    close(fhandle)
    return(ret)
}

File: posted/read.py
from pathlib import Path

import pandas as pd
import yaml


def read_tedf_from_csv(fpath: str):
    """
    Read CSV data file

    Parameters
    ----------
    fpath: str
        Path of the file to read
    Returns
    -------
        pd.DataFrame
            DataFrame containing the data of the CSV
    """
    df = pd.read_csv(
        fpath,
        sep=",",
        quotechar='"',
        encoding="utf-8",
        dtype=str,
    ).fillna("")
    df.index = df.index + 2
    return df


def read_yaml(fpath: Path):
    """
    Read YAML config file

    Parameters
    ----------
    fpath: str
        Path of the file to read
    Returns
    -------
        dict
            Dictionary containing config
    """
    with open(fpath, mode="r", encoding="utf-8") as file_handle:
        return yaml.load(
            stream=file_handle,
            Loader=yaml.FullLoader,
        )
